test-cypress/cypress/e2e/2-advanced-examples/spies_stubs_clocks.cy.js

```diff
@@ -0,0 +1,204 @@
+/// <reference types="cypress" />
+
+context('Spies, Stubs, and Clock', () => {
+  it('cy.spy() - wrap a method in a spy', () => {
+    // https://on.cypress.io/spy
+    cy.visit('https://example.cypress.io/commands/spies-stubs-clocks');
+
+    const obj = {
+      foo () {}
+    };
+
+    const spy = cy.spy(obj, 'foo').as('anyArgs');
+
+    obj.foo();
+
+    expect(spy).to.be.called;
+  });
+
+  it('cy.spy() retries until assertions pass', () => {
+    cy.visit('https://example.cypress.io/commands/spies-stubs-clocks');
+
+    const obj = {
+      /**
+       * Prints the argument passed
+       * @param x {any}
+      */
+      foo (x) {
+        console.log('obj.foo called with', x);
+      }
+    };
+
+    cy.spy(obj, 'foo').as('foo');
+
+    setTimeout(() => {
+      obj.foo('first');
+    }, 500);
+
+    setTimeout(() => {
+      obj.foo('second');
+    }, 2500);
+
+    cy.get('@foo').should('have.been.calledTwice');
+  });
+
+  it('cy.stub() - create a stub and/or replace a function with stub', () => {
+    // https://on.cypress.io/stub
+    cy.visit('https://example.cypress.io/commands/spies-stubs-clocks');
+
+    const obj = {
+      /**
+       * prints both arguments to the console
+       * @param a {string}
+       * @param b {string}
+      */
+      foo (a, b) {
+        console.log('a', a, 'b', b);
+      }
+    };
+
+    const stub = cy.stub(obj, 'foo').as('foo');
+
+    obj.foo('foo', 'bar');
+
+    expect(stub).to.be.called;
```

Comment:
The assertion `expect(stub).to.be.called;` might not work as expected because it should be `expect(stub).to.have.been.called;` to properly check if the stub was called.